resources/prosody-plugins/mod_av_moderation_component.lua

```diff
@@ -0,0 +1,242 @@
+            local stanza = st.message({
+                from = module.host; to = occupant.jid; }):tag('json-message',{ xmlns = 'http://jitsi.org/jitmeet' })
+                    :text(body_to_send):up();
+            module:send(stanza);
+        end
+    end
+end
+
+-- receives messages from clients to the component sending A/V moderation enable/disable commands or adding
+-- jids to the whitelist
+function on_message(event)
+    local session = event.origin;
+
+    -- Check the type of the incoming stanza to avoid loops:
+    if event.stanza.attr.type == 'error' then
+        return; -- We do not want to reply to these, so leave.
+    end
+
+    if not session or not session.jitsi_web_query_room then
+        return false;
+    end
+
+    local moderation_command = event.stanza:get_child('av_moderation');
+
+    if moderation_command then
+        -- get room name with tenant and find room
+        local room = get_room_by_name_and_subdomain(session.jitsi_web_query_room, session.jitsi_web_query_prefix);
+
+        if not room then
+            module:log('warn', 'No room found found for %s/%s',
+                    session.jitsi_web_query_prefix, session.jitsi_web_query_room);
+            return false;
+        end
+
+        -- check that the participant requesting is a moderator and is an occupant in the room
+        local from = event.stanza.attr.from;
+        local occupant = room:get_occupant_by_real_jid(from);
+        if not occupant then
+            log('warn', 'No occupant %s found for %s', from, room.jid);
+            return false;
+        end
+        if occupant.role ~= 'moderator' then
+            log('warn', 'Occupant %s is not moderator and not allowed this operation for %s', from, room.jid);
+            return false;
+        end
+
+        local mediaType = moderation_command.attr.mediaType;
+        if mediaType then
+            if mediaType ~= 'audio' and mediaType ~= 'video' then
+                module:log('warn', 'Wrong mediaType %s for %s', mediaType, room.jid);
+                return false;
+            end
+        else
+            module:log('warn', 'Missing mediaType for %s', room.jid);
+            return false;
+        end
+
+        if moderation_command.attr.enable ~= nil then
+            local enabled;
+            if moderation_command.attr.enable == 'true' then
+                enabled = true;
+                if room.av_moderation and room.av_moderation[mediaType] then
+                    module:log('warn', 'Concurrent moderator enable/disable request or something is out of sync');
+                else
+                    room.av_moderation = {};
+                    room.av_moderation_actors = {};
+                    room.av_moderation[mediaType] = {};
+                    room.av_moderation_actors[mediaType] = occupant.nick;
+                end
+            else
+                enabled = false;
+                if not room.av_moderation and not room.av_moderation[mediaType] then
```

Comment:
s/and/or/? I thnk as it is the condition is always false

Reply:
Same as above -- should notfications be sent if the state was not changed?

Reply:
Fixed.